packages/core/database/lib/entity-manager/relations-orderer.js

```diff
@@ -87,17 +162,19 @@ const relationsOrderer = (initArr, idColumn, orderColumn) => {
       return this;
     },
     connect(relations) {
-      _.castArray(relations).forEach((relation) => {
+      const sortedRelations = sortConnectArray(relations, arr);
+      sortedRelations.forEach((relation) => {
         this.disconnect(relation);
 
         try {
           insertRelation(relation);
         } catch (err) {
-          strapi.log.error(err);
           throw new Error(
-            `Could not connect ${relation.id}, position ${JSON.stringify(
+            `There was a problem connecting relation with id ${
+              relation.id
+            } at position ${JSON.stringify(
               relation.position
-            )} is invalid`
+            )}. The list of connect relations is not valid`
```

Comment:
Yes, it will be looking for a relation that does not exist.

But yes, maybe the sort connect array should do it, it would be clearer when tracking the error.